core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java

```diff
@@ -71,7 +71,7 @@ public void testSubtract()
         assertFunction("100000000037 - 37", BIGINT, 100000000037L - 37L);
         assertFunction("37 - 100000000017", BIGINT, 37 - 100000000017L);
         assertFunction("100000000017 - 37", BIGINT, 100000000017L - 37L);
-        assertFunction("100000000017 - 100000000017", BIGINT, 100000000017L - 100000000017L);
+        assertFunction("100000000017 - 100000000017", BIGINT, 0L);
```

Comment:
That's a false positive in a sense that the original code was intentionally written this way.
I don't think it's a big issue, but worth noting.

Reply:
We can have this as a comment?

Reply:
In my series adding Error Prone checks, I just added a suppression for this, as I recognized that this was intentional:
https://github.com/trinodb/trino/pull/6879/files#diff-2fec98bebd272276c2c0cafe53035760f5a45e6cc80f16be071aeecac6dcbf76R69

Reply:
Is it a question to me?

Reply:
(The commit from the link above is part of a longer series which I started to split into smaller pieces, I haven't gotten to this one yet)

Reply:
> Is it a question to me?

To audience (author and other reviewers).

Reply:
@ksobolew suppressions are a signal that check may not be helpful overall.

Reply:
Yes, I agree but this change doesn't make code less readable imo

Reply:
@findepi yeah, I agree, but there are always exceptions and oddball cases, like this one. Ideally there should be a comment why the suppression is warranted, though, and I acknowledge that I didn't do that in my commit